File: TeeNew/Tool_PageNumDemo.pas
unit Tool_PageNumDemo;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeEdiGene, Series, TeEngine, TeeProcs, Chart, TeePageNumTool,
  TeeTools, TeeNavigator;

type
  TPageNumToolDemo = class(TBaseForm)
    Series1: TPointSeries;
    Series2: TLineSeries;
    Label1: TLabel;
    Edit1: TEdit;
    Label2: TLabel;
    ComboBox1: TComboBox;
    ChartPageNavigator1: TChartPageNavigator;
    Label3: TLabel;
    Button1: TButton;
    procedure FormCreate(Sender: TObject);
    procedure Edit1Change(Sender: TObject);
    procedure ComboBox1Change(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
    PageTool : TPageNumTool;
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses EditChar, TeeBrushDlg;

procedure TPageNumToolDemo.FormCreate(Sender: TObject);
begin
  inherited;
  Chart1.SeriesList.FillSampleValues(50);

  // Create page num tool:
  PageTool:=TPageNumTool.Create(Self);
  PageTool.ParentChart:=Chart1;
  PageTool.Shape.Font.Color:=clRed;
  PageTool.Shape.Shadow.Size:=1;
  PageTool.Shape.Shadow.Color:=clDkGray;

  Edit1.Text:=PageTool.Format;
end;

procedure TPageNumToolDemo.Edit1Change(Sender: TObject);
begin
  PageTool.Format:=Edit1.Text;
end;

procedure TPageNumToolDemo.ComboBox1Change(Sender: TObject);
begin
  PageTool.Position:=TAnnotationPosition(ComboBox1.ItemIndex);
end;

procedure TPageNumToolDemo.Button1Click(Sender: TObject);
begin
  EditChartTool(Self,PageTool);
end;

procedure TPageNumToolDemo.FormShow(Sender: TObject);
begin
  inherited;
  ComboBox1.ItemIndex:=0;
end;

initialization
  RegisterClass(TPageNumToolDemo);
end.

File: TeeNew/Builder2009/Tool_PageNumDemo.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "Tool_PageNumDemo.h"
#include "EditChar.hpp"
#include "TeeBrushDlg.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeNavigator"
#pragma link "TeeEdiGene"
#pragma resource "*.dfm"

TPageNumToolDemo *PageNumToolDemo;
TTeeCustomToolClass toolclass[1] = {__classid(TPageNumTool)};

//---------------------------------------------------------------------------
__fastcall TPageNumToolDemo::TPageNumToolDemo(TComponent* Owner)
        : TBaseForm(Owner)
{
}
__fastcall TPageNumToolDemo::~TPageNumToolDemo(void)
{
}
//---------------------------------------------------------------------------
void __fastcall TPageNumToolDemo::FormCreate(TObject *Sender)
{
  // RegisterTeeTools(toolclass,0);
  Series2->FillSampleValues(50);
  Series1->FillSampleValues(50);

  PageTool = new TPageNumTool(this);
  PageTool->ParentChart = Chart1;
  PageTool->Shape->Font->Color = clRed;
  PageTool->Shape->ShadowSize = 1;
  PageTool->Shape->ShadowColor = clDkGray;

  Edit1->Text = PageTool->Format;
}
//---------------------------------------------------------------------------

void __fastcall TPageNumToolDemo::Edit1Change(TObject *Sender)
{
  PageTool->Format = Edit1->Text;
}
//---------------------------------------------------------------------------

void __fastcall TPageNumToolDemo::ComboBox1Change(TObject *Sender)
{
  PageTool->Position = TAnnotationPosition(ComboBox1->ItemIndex);
}
//---------------------------------------------------------------------------

void __fastcall TPageNumToolDemo::Button1Click(TObject *Sender)
{
  EditChartTool(this,PageTool);
}
//---------------------------------------------------------------------------

void __fastcall TPageNumToolDemo::FormShow(TObject *Sender)
{
  ComboBox1->ItemIndex = 0;
}
//---------------------------------------------------------------------------


void __fastcall TPageNumToolDemo::FormDestroy(TObject *Sender)
{
  // UnRegisterTeeTools(toolclass,0);
}
//---------------------------------------------------------------------------

